core/src/test/java/dk/alexandra/fresco/lib/lp/LpBuildingBlockTests.java

```diff
@@ -0,0 +1,602 @@
+      // for (int i = 0; i < b.length; i++) {
+      // updatedB[i] = innerProduct(b, updateMatrix.getIthRow(i));
+      // updatedColumn[i] = innerProduct(column, updateMatrix.getIthRow(i));
+      // }
+      // int exitingIndex = 0;
+      // BigInteger half = mod.divide(BigInteger.valueOf(2));
+      // BigInteger minNominator = null;
+      // BigInteger minDenominator = null;
+      // for (int i = 0; i < updatedB.length; i++) {
+      // boolean nonPos = updatedColumn[i].compareTo(half) > 0;
+      // nonPos = nonPos || updatedColumn[i].compareTo(BigInteger.ZERO) == 0;
+      // if (!nonPos) {
+      // if (minNominator == null) {
+      // minNominator = updatedB[i];
+      // minDenominator = updatedColumn[i];
+      // exitingIndex = i;
+      // } else {
+      // BigInteger leftHand = minNominator.multiply(updatedColumn[i]);
+      // BigInteger rightHand = minDenominator.multiply(updatedB[i]);
+      // BigInteger diff = leftHand.subtract(rightHand).mod(mod);
+      // diff = diff.compareTo(half) > 0 ? diff.subtract(mod) : diff;
+      // if (diff.compareTo(BigInteger.ZERO) > 0) {
+      // minNominator = updatedB[i];
+      // minDenominator = updatedColumn[i];
+      // exitingIndex = i;
+      // }
+      // }
+      // }
+      // }
+      // return exitingIndex;
+      return 0;
+    }
+
+    // private BigInteger innerProduct(BigInteger[] a, BigInteger[] b) {
+    // if (a.length > b.length) {
+    // throw new RuntimeException("b vector too short");
+    // }
+    // BigInteger result = BigInteger.valueOf(0);
+    // for (int i = 0; i < a.length; i++) {
+    // result = (result.add(a[i].multiply(b[i]))).mod(mod);
+    // }
+    // return result;
+    // }
+  }
+
+  public static class TestLpTableuDebug<ResourcePoolT extends ResourcePool>
+      extends TestThreadFactory<ResourcePoolT, ProtocolBuilderNumeric> {
+
+    ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+    PrintStream stream = new PrintStream(bytes);
+
+    public TestLpTableuDebug() {}
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      return new TestThread<ResourcePoolT, ProtocolBuilderNumeric>() {
+
+        @Override
+        public void test() throws Exception {
+          LpTabluauTester app = new LpTabluauTester() {
+
+            @Override
+            public DRes<Object> buildComputation(ProtocolBuilderNumeric builder) {
+              setup(builder, stream);
+              return () -> null;
+            }
+          };
+          runApplication(app);
+          String output = bytes.toString("UTF-8");
+          System.out.println(output);
+          assertTrue(output
+              .contains("C: \n" + "0, 0, 0, \n" + "1, 1, 1, \n" + "2, 2, 2, \n" + "3, 3, 3,"));
+          assertTrue(output.contains("B: \n" + "10, 10, 10, 10,"));
+          assertTrue(output.contains("F: \n" + "0, 2, 3,"));
+          assertTrue(output.contains("z: \n" + "0"));
```

Comment:
Nice test, maybe recode it so future work on the logging only needs to change a single line

Reply:
Not sure what you are thinking here. Just put everything in one assertTrue?